### PR TITLE
Workaround bug on logic op with float framebuffer

### LINKED ISSUE #6859: [Bug]Paper Mario TTYD black screen
### Description of the issue

Same problem with #6851 , but my GPU is Intel.
AMD's black screen has been fixed, but Intel has not.
(Google Translate)
QQ截图20240523221112

### Reproduction steps

Black screen after first entry into sewer
(Google Translate)

### Log file

[Ryujinx_1.1.1320_2024-05-23_18-51-25.log](https://github.com/Ryujinx/Ryujinx/files/15417984/Ryujinx_1.1.1320_2024-05-23_18-51-25.log)

### OS

Windows 11 23H2 22631.3374

### Ryujinx version

1.1.1320

### Game version

1.0.0

### CPU

i5-12400

### GPU

Intel(R) UHD Graphics 730 (31.0.101.5522)

### RAM

16GB

### List of applied mods

_No response_

### Additional context?

_No response_

### PR DESCRIPTION
**Pull request by piplup55**
built on top of #6852 to add support for intel gpu's
NOTE: this won't fix the crashing on windows

fixes #6859

before:
image

after:
image

### CODE REVIEW

**Comment by gdkchan** [comment]
Maybe it's better to enable it for all GPUs that are not NVIDIA.

**Comment by piplup55** [comment]
> Maybe it's better to enable it for all GPUs that are not NVIDIA.

so we would enable it for nvidia only, i guess that makes more sense, i'm unware if it affects any other vendors

so something like this
```bool logicOpEnable = LogicOpEnable && (gd.Vendor == Vendor.Nvidia || Internal.LogicOpsAllowed);```
instead of
```bool logicOpEnable = LogicOpEnable && (gd.Vendor != Vendor.Amd && gd.Vendor != Vendor.Intel || Internal.LogicOpsAllowed);```

**Comment by gdkchan** [comment]
> > Maybe it's better to enable it for all GPUs that are not NVIDIA.
> 
> so we would enable it for nvidia only, i guess that makes more sense, i'm unware if it affects any other vendors
> 
> so something like this `bool logicOpEnable = LogicOpEnable && (gd.Vendor == Vendor.Nvidia || Internal.LogicOpsAllowed);` instead of `bool logicOpEnable = LogicOpEnable && (gd.Vendor != Vendor.Amd && gd.Vendor != Vendor.Intel || Internal.LogicOpsAllowed);`

Yes. Most of the other vendors are mobile vendors that don't even support logic ops at all. There's newer Adreno GPUs that supports it I think, and Samsung Xclipse GPUs, but I guess those would have the bug too since they are based on AMD RDNA architecture.

**Comment by piplup55** [comment]
> Yes. Most of the other vendors are mobile vendors that don't even support logic ops at all. There's newer Adreno GPUs that supports it I think, and Samsung Xclipse GPUs, but I guess those would have the bug too since they are based on AMD RDNA architecture.

i'll make the change and test it on amd and intel, it should work in theory but just in case

**Comment by Ozzel05** [comment]
Im a little unsure on how to install this as im a switch emu rookie. How do I?

File: src/Ryujinx.Graphics.Vulkan/PipelineState.cs [hunk 1]
```diff
@@ -560,9 +560,9 @@ public void Initialize()
                     }
                 }
 
-                // AMD has a bug where it enables logical operations even for float formats,
+                // vendors other than nvidia have a bug where it enables logical operations even for float formats,
```
Comment (gdkchan) [inline src/Ryujinx.Graphics.Vulkan/PipelineState.cs:563]:
```suggestion
                // Vendors other than NVIDIA have a bug where it enables logical operations even for float formats,
```

**Review by gdkchan — APPROVED** [review]
lgtm, minus comment.

**Review by riperiperi — APPROVED** [review]

**Comment by R-Cal1** [comment]
> Download the artifacts for this pull request:
> 
>     * [ryujinx-Release-1.1.0+ff69625-linux_arm64](https://nightly.link/Ryujinx/Ryujinx/actions/artifacts/1532152507.zip)
> 
>     * [ryujinx-Release-1.1.0+ff69625-linux_x64](https://nightly.link/Ryujinx/Ryujinx/actions/artifacts/1532157267.zip)
> 
>     * [ryujinx-Release-1.1.0+ff69625-macos_universal](https://nightly.link/Ryujinx/Ryujinx/actions/artifacts/1532160797.zip)
> 
>     * [ryujinx-Release-1.1.0+ff69625-win_x64](https://nightly.link/Ryujinx/Ryujinx/actions/artifacts/1532170392.zip)
> 
> 
> Old GUI (GTK3)
> 
> 
> GUI-less (SDL2)
> 
> 
> Only for Developers

Hi, the links are broken, is it possible to fix them ? Thanks

**Comment by piplup55** [comment]
this was merged awhile ago, the changes are already in master

**Pull request merged**